src/aks-preview/azext_aks_preview/_params.py

```diff
@@ -409,6 +409,7 @@ def load_arguments(self, _):
         c.argument('enable_secret_rotation', action='store_true')
         c.argument('rotation_poll_interval')
         c.argument('enable_sgxquotehelper', action='store_true')
+        c.argument('enable_app_routing', options_list=['--enable-app-routing'], action='store_true', is_preview=True)
```

Comment:
There is no need to explicitly declare the default option name
```suggestion
        c.argument('enable_app_routing', action='store_true', is_preview=True)
```